lib/reverse_adoc/converters/pre.rb

```diff
@@ -1,17 +1,19 @@
 module ReverseAdoc
   module Converters
     class Pre < Base
-      def convert(node, state = {})
+      def to_coradoc(node, state = {})
         id = node['id']
-        anchor = id ? "[[#{id}]]\n" : ""
         lang = language(node)
         content = treat_children(node, state)
         if lang
-          "\n\n#{anchor}[source,#{lang}]\n----\n" << content.lines.to_a.join("") << "\n----\n\n"
+          Coradoc::Document::Block.new(nil, lines: content.lines, delimiter: '----', lang: lang, id: id)
```

Comment:
`<pre>` should lead:
* a `Literal` block
* if it has a `lang` attribute, make it a `SourceCode` block.

There's no need to give any delimiter for either.